tools/serverpod_cli/test/generator/dart/server_code_generator/database_repository_class_test.dart

```diff
@@ -0,0 +1,453 @@
+            'The class ${testClassName}DetachRepository was found but was expected to not exist.',
+      );
+    });
+  });
+  group(
+      'Given a class with table name and object relation field when generating code',
+      () {
+    var entities = [
+      ClassDefinitionBuilder()
+          .withClassName(testClassName)
+          .withFileName(testClassFileName)
+          .withTableName('example_table')
+          .withObjectRelationField(
+            'company',
+            'Company',
+            'company',
+            nullableRelation: true,
+          )
+          .withObjectRelationField(
+            'address',
+            'Address',
+            'address',
+            foreignKeyFieldName: 'companyAddressId',
+          )
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    var compilationUnit = parseString(content: codeMap[expectedFilePath]!).unit;
+
+    var repositoryClass = CompilationUnitHelpers.tryFindClassDeclaration(
+      compilationUnit,
+      name: '${testClassName}Repository',
+    );
+
+    group('then the class name ${testClassName}Repository', () {
+      test('has a final attach field', () {
+        var field = CompilationUnitHelpers.tryFindFieldDeclaration(
+          repositoryClass!,
+          name: 'attach',
+        );
+
+        expect(
+          field?.toSource(),
+          'final attach = const ${testClassName}AttachRepository._();',
+          reason: 'Missing static instance field.',
+        );
+      });
+
+      test('has a final detach field', () {
+        var field = CompilationUnitHelpers.tryFindFieldDeclaration(
+          repositoryClass!,
+          name: 'detach',
+        );
+
+        expect(
+          field?.toSource(),
+          'final detach = const ${testClassName}DetachRepository._();',
+          reason: 'Missing static instance field.',
+        );
+      });
+    }, skip: repositoryClass == null);
+
+    test('then a class named ${testClassName}AttachRepository is generated',
+        () {
+      expect(
+        CompilationUnitHelpers.hasClassDeclaration(
+          compilationUnit,
+          name: '${testClassName}AttachRepository',
+        ),
+        isTrue,
+        reason:
+            'Expected the class ${testClassName}AttachRepository to be generated.',
+      );
+    });
+
+    var repositoryAttachClass = CompilationUnitHelpers.tryFindClassDeclaration(
+      compilationUnit,
+      name: '${testClassName}AttachRepository',
+    );
+
+    group('then the ${testClassName}AttachRepository', () {
+      test('has a private constructor', () {
+        var constructor = CompilationUnitHelpers.tryFindConstructorDeclaration(
+          repositoryAttachClass!,
+          name: '_',
+        );
+        expect(
+          constructor?.toSource(),
+          'const ${testClassName}AttachRepository._();',
+          reason: 'Missing private constructor.',
+        );
```

Comment:
I think this type of match is better as you get way more helpful information when a test fails, and you are still checking the exact same thing. 

Not sure what you mean by adding the toSource to the compilation unit helpers? We cannot mix bool and string right? And if you do not have a match at all you would have to output the entire source file rather than the specific method declaration we are checking. By scoping down the match we get a much clearer error when things fails.